src/winerror.cr

```diff
@@ -0,0 +1,2104 @@
+class WinError < Exception
```

Comment:
For now mapping to errno will work but we *must* remove errno before the first release. We should abstract the exception hierarchy to be platform specific. There's a good document on how python did this that we can take inspiration from. Doing that is beyond the scope of this or though.